searx/engines/yahoo.py

```diff
@@ -157,14 +183,22 @@ def response(resp):
     results = []
     dom = html.fromstring(resp.text)
 
+    url_xpath = './/h3/a/@href'
```

Comment:
```suggestion
    url_xpath = './/div[contains(@class,"compTitle")]/h3/a/@href'
```
The `compTitle` class is also used everywhere for the localized pages from my tests, so it might be more robust to add it here too. The thing that's mainly different is the ordering of `h3 > a` vs `a > h3`